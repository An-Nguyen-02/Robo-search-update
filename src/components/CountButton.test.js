import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CountButton from './CountButton';

it('button render snapshot',()=>{
	const mockColor='red';
	expect(shallow(<CountButton color={mockColor} />)).toMatchSnapshot();
})

it('correct increment',()=>{
	const mockColor='red';
	const wrapper = shallow(<CountButton color={mockColor} />);
	wrapper.find('[id="counter"]').simulate('click')
	expect(wrapper.state()).toEqual({count:1})
	wrapper.find('[id="counter"]').simulate('click')
	wrapper.find('[id="counter"]').simulate('click')
	expect(wrapper.state()).toEqual({count:3})
	wrapper.find('[id="counter"]').simulate('keypress')
	expect(wrapper.state()).toEqual({count:3})
	expect(wrapper.props().color).toEqual('red')
})