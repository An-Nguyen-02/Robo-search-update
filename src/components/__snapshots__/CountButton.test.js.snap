// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render snapshot 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
